### PR TITLE
Added check for duplicate values in dropdown

### PR DESCRIPTION
**Pull request by Caitlin-Fotheringham**
Added check for any duplicates, returning invalid if any are present.

Fixes appsmithorg/appsmith#655

### CODE REVIEW

File: app/client/src/utils/Validators.ts [hunk 1]
```diff
@@ -372,12 +373,13 @@ export const VALIDATORS: Record<ValidationType, Validator> = {
...
     const hasOptions = _.every(parsed, (datum: { label: any; value: any }) => {
       if (_.isObject(datum)) {
+        if(_.isString(datum.value)) values.push(datum.value)
         return isValidOption(datum);
       } else {
         return false;
       }
     });
-    if (!hasOptions) {
+    if (!hasOptions || new Set(values).size !== values.length) {
```
Comment (hetunandu) [inline app/client/src/utils/Validators.ts:382]: can we instead use the unique function of lodash and return a parsed value what is always guaranteed to be unique

**Review by hetunandu — CHANGES_REQUESTED** [review]
Great start @Caitlin-Fotheringham ! But we always want to return options even if one of them is invalid. That means that we need to make sure that even if there are duplicate `option.value` we will parse and return a unique list of options from here

File: app/client/src/utils/Validators.ts (hunk 1 above)
Comment (Caitlin-Fotheringham) [reply app/client/src/utils/Validators.ts:382]: I will work on this today! To clarify, if a duplicate is found do you want it to return as invalid or do you want to ensure parsed all has unique values, removing any duplicates, but returning as valid?

Comment (hetunandu) [reply app/client/src/utils/Validators.ts:382]: If duplicates are found, mark it as invalid but return unique options in `parsed`

Comment (Caitlin-Fotheringham) [reply app/client/src/utils/Validators.ts:382]: I see what you mean now, thank you

File: app/client/src/utils/Validators.ts [hunk 2]
```diff
@@ -372,15 +373,17 @@ export const VALIDATORS: Record<ValidationType, Validator> = {
...
       if (_.isObject(datum)) {
+        if(_.isString(datum.value)) values.push(datum.value)
         return isValidOption(datum);
       } else {
         return false;
       }
     });
-    if (!hasOptions) {
+
+    if (!hasOptions || _.uniq(values).length !== values.length) {
```
Comment (hetunandu) [inline app/client/src/utils/Validators.ts:383]: Why do we create a new values array over here? Can't we just compare the length of `parsed` and `_.uniq(parsed, 'value')`

Comment (Caitlin-Fotheringham) [reply app/client/src/utils/Validators.ts:383]: That is a very good point, I'll do that now

**Review by hetunandu — APPROVED** [review]

**Comment by hetunandu** [comment]
Thanks for this @Caitlin-Fotheringham ! This was a good PR. 😃 Hope to see more contributions from you in the coming days

**Pull request merged**